src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -1043,14 +1077,26 @@ void GoogleCalendarSyncAdaptor::updateLocalCalendarNotebookEvents(int accountId,
 
             m_idDb.removeEvent(accountId, eventId);
             if (allMap.contains(eventId)) {
+                // currently existing base event or persistent occurrence which needs deletion
+                SOCIALD_LOG_DEBUG("Event deleted remotely:" << eventId);
                 m_calendar->deleteEvent(allMap.value(eventId));
                 m_storageNeedsSave = true;
+            } else if (allMap.contains(parentId)) {
+                // this is a non-persistent occurrence, we need to add an EXDATE to the base event.
+                SOCIALD_LOG_DEBUG("Occurrence deleted remotely:" << eventId << "for recurrenceId:" << recurrenceId.toString());
+                KCalCore::Event::Ptr event = allMap.value(parentId);
+                event->startUpdates();
+                event->recurrence()->addExDateTime(recurrenceId);
```

Comment:
Yes, the Google API is a bit strange here.  This case is where you have a recurring series on Google.  There are NO persistent occurrences / exceptions, ie all of the recurrances are just RRULE occurrences.  Then in the web UI you delete one of those occurrences.

What I would expect is that Google would just send the original series event as modified, with an EXDATE added.  Unfortunately, this is not what it does: instead, it sends an event with status=CANCELLED which is formulated as if it were a persistent occurrence (ie, it has a RECURRENCE-ID set it in, which corresponds to a specific RDATE/RRULE occurrence).

So, this branch basically detects this case (a CANCELLED event which corresponds to a normal recurrance of a series event) and does the appropriate thing (adds an exdate to that series event).